pkg/roachprod/install/cockroach.go

```diff
@@ -520,17 +520,27 @@ func (c *SyncedCluster) NodeURL(
 }
 
 // NodePort returns the system tenant's SQL port for the given node.
-func (c *SyncedCluster) NodePort(ctx context.Context, node Node) (int, error) {
-	desc, err := c.DiscoverService(ctx, node, SystemInterfaceName, ServiceTypeSQL, 0)
+func (c *SyncedCluster) NodePort(
+	ctx context.Context, node Node, virtualClusterName string, sqlInstance int,
+) (int, error) {
+	if virtualClusterName == "" {
```

Comment:
I think `DiscoverServices` already has a check for this, https://github.com/cockroachdb/cockroach/blob/master/pkg/roachprod/install/services.go#L142